pontis-ui/src/components/Navigation/CommandPallete.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import classNames from "classnames";
+import { Dialog, Combobox, Transition } from "@headlessui/react";
+
+import { AssetKeys } from "../../hooks/oracle";
+import { ChevronRightIcon, SearchIcon } from "@heroicons/react/outline";
+import AssetCardName from "../Asset/AssetCardName";
+import { useSearch } from "../../providers/search";
+import { assetKeyToUrl } from "../../../utils/encodeUrl";
+
+interface CommandPallateProps {
+  isOpen: boolean;
+}
+
+const CommandPallate: React.FC<CommandPallateProps> = ({ isOpen }) => {
+  const router = useRouter();
+  const setSearch = useSearch();
+  const [query, setQuery] = useState("");
+
+  useEffect(() => {
+    /**
+     * Listens for shortcut to open command pallete.
+     * @param {KeyboardEvent} event keydown event
+     */
+    function onKeydown(event) {
+      if (event.key === "k" && (event.metaKey || event.ctrlKey)) {
+        // AFAIK (isOpen) => !isOpen instead of !isOpen allows us to remove isOpen from the dependency array.
+        // This prevents us from mounting and unmounting the event listeners on every render.
+        setSearch((isOpen) => !isOpen);
+      }
+    }
+    window.addEventListener("keydown", onKeydown);
+    return () => {
+      window.removeEventListener("keydown", onKeydown);
+    };
+  }, [setSearch]);
+
+  const filteredAssets = query
+    ? AssetKeys.filter((asset) =>
```

Comment:
Hmm this kind of depends on what behavior we want.

1. We want all assets to be shown when you open the search modal. Then we should do your simplification. This can cause a little flash of assets when closing the modal after typing the name of a specific asset. (not 100% sure why)
2. We want to assets to be shown. Then, my solution works, although I don't know why it evaluates to false when `query = ""`. So, we could make the check more explicit.

Reply:
I think it is helpful to show assets even when the search query is empty, as otherwise that real estate is wasted. Similar functionality elsewhere does this too, e.g. searching on telegram. Not sure why the flickering happens, likely due to race condition between state reset and removing the component from the DOM?